### PR TITLE
Add org.sugarlabs.AbacusActivity

### PR DESCRIPTION
**Pull request by tchx84**
Abacus lets the learner explore different representations of numbers using different mechanical counting systems developed by the ancient Romans and Chinese. There are several different variants available for exploration.

### CODE REVIEW

**Comment by tchx84** [comment]
@nedrichards can you take a look?

**Comment by tchx84** [comment]
bot, build org.sugarlabs.AbacusActivity

**Comment by tchx84** [comment]
> Ah, I should've said when reviewing in person, but we can't have the sugar shared modules stuff in a separate repo. You'll need to add it in this PR for review. I know that there's a bunch of these upcoming, so I'd be pro moving them to the flathub shared-modules when we've landed a few - but for now can we have it in here?

Sure, I can do that.

> 
> I'm also still slightly uncomfortable with forcing the x11 GDK backend. Did you test that on wayland whilst removing the permissions for the X11 socket?
> 

I tried removing x11 socket and, of course, it wouldn't work/run.

Removing it is problematic, without the x11 backend sugar won't properly position its "palettes/popups", because its implementation is very x11 dependent. Would this be a blocker? I checked that there are a couple other apps doing the same (e.g. com.uploadedlobster.peek).

> But other than that LGTM

Well, at least the checks passed haha...

**Comment by bilelmoussaoui** [comment]
It's probably worth using a BaseApp later if it's worth it instead of a having those suagar specific modules in shared-modules

File: org.sugarlabs.AbacusActivity.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ],
+    "modules": [
+        "sugar-shared-modules/sugar/artwork.json",
+        "sugar-shared-modules/sugar/toolkit-gtk3.json",
+        "sugar-shared-modules/python/sugarapp.json",
+        {
+            "name": "abacus",
+            "buildsystem": "simple",
+            "build-commands": [
+                "python${PYTHON_VERSION} setup.py install --prefix=${FLATPAK_DEST} --skip-install-desktop-file --skip-install-mime"
```
Comment (bilelmoussaoui) [inline org.sugarlabs.AbacusActivity.json]: Any reason for doing this? We don't have python2 anymore in the runtime, you can use python3

**Comment by nedrichards** [comment]
It's not a blocker to be X11 only at all. Just remove the wayland socket and the GDK backend env var. Then fix the wayland support upstream in a future release and add it then. For now 100% of wayland flatpak users have access to XWayland so it's not the end of the world.

A baseapp might work, but my understanding was that there was a bit of mix and match in the sugar apps, they didn't all need everything.

**Comment by nedrichards** [comment]
In manual testing on Endless I had issues unless I added --nosocket=wayland so *certainly* remove that. Works fine other than that though.

File: org.sugarlabs.AbacusActivity.json (hunk 1 above)
Comment (tchx84) [reply org.sugarlabs.AbacusActivity.json]: Not in this specific case (I will remove it) but, for the dependencies, Sugar is currently porting from python2 to python3, so there is a huge mix between 2 and 3 apps. I did this for not having to duplicate all these dependency files.

**Comment by tchx84** [comment]
> It's not a blocker to be X11 only at all. Just remove the wayland socket and the GDK backend env var. Then fix the wayland support upstream in a future release and add it then. For now 100% of wayland flatpak users have access to XWayland so it's not the end of the world.
> 

Alright.

> A baseapp might work, but my understanding was that there was a bit of mix and match in the sugar apps, they didn't all need everything.

Yes, they definitely don't need everything.

**Comment by tchx84** [comment]
> In manual testing on Endless I had issues unless I added --nosocket=wayland so _certainly_ remove that. Works fine other than that though.

Alright.

**Comment by tchx84** [comment]
bot, build org.sugarlabs.AbacusActivity

**Comment by tchx84** [comment]
@nedrichards @bilelmoussaoui all comment addressed!

**Comment by tchx84** [comment]
err, forgot to add the specific commits to some of these dependencies ! I am boarding my flight back to Paraguay, but will do ASAP.

**Comment by nedrichards** [comment]
I think you're missing the toolbar icon for 'binary'

**Comment by tchx84** [comment]
@nedrichards @bilelmoussaoui I have made the changes so the build is properly reproducible now.

**Comment by tchx84** [comment]
bot, build org.sugarlabs.AbacusActivity

**Comment by tchx84** [comment]
> I think you're missing the toolbar icon for 'binary'

Can you run this last build and upload an screenshot ? I can't reproduce that, I tried in different screen resolutions.

I made some changes related to how the fullscreen is setup now, and fixed some artifacts for me, it should be unrelated, but I am curious it would make any different for this case as well.

**Comment by tchx84** [comment]
bot, build org.sugarlabs.AbacusActivity

**Comment by tchx84** [comment]
Added a few cleanups to the manifest.

**Pull request closed without merging**